Source/Core/VideoCommon/VertexLoader.cpp

```diff
@@ -312,38 +306,42 @@ void VertexLoader::CompileVertexTranslator()
   }
 
   // Texture matrix indices (remove if corresponding texture coordinate isn't enabled)
-  for (int i = 0; i < 8; i++)
+  for (size_t i = 0; i < m_VtxDesc.high.TexCoord.Size(); i++)
   {
     m_native_vtx_decl.texcoords[i].offset = nat_offset;
     m_native_vtx_decl.texcoords[i].type = VAR_FLOAT;
     m_native_vtx_decl.texcoords[i].integer = false;
 
-    const int format = m_VtxAttr.texCoord[i].Format;
-    const int elements = m_VtxAttr.texCoord[i].Elements;
+    const auto format = m_VtxAttr.texCoord[i].Format;
+    const auto elements = m_VtxAttr.texCoord[i].Elements;
 
-    if (tc[i] != NOT_PRESENT)
+    if (m_VtxDesc.high.TexCoord[i] != VertexComponentFormat::NotPresent)
     {
-      ASSERT_MSG(VIDEO, DIRECT <= tc[i] && tc[i] <= INDEX16,
-                 "Invalid texture coordinates!\n(tc[i] = %d)", (u32)tc[i]);
-      ASSERT_MSG(VIDEO, FORMAT_UBYTE <= format && format <= FORMAT_FLOAT,
-                 "Invalid texture coordinates format!\n(format = %d)", format);
-      ASSERT_MSG(VIDEO, 0 <= elements && elements <= 1,
-                 "Invalid number of texture coordinates elements!\n(elements = %d)", elements);
+      ASSERT_MSG(VIDEO,
+                 VertexComponentFormat::NotPresent <= m_VtxDesc.high.TexCoord[i] &&
+                     m_VtxDesc.high.TexCoord[i] <= VertexComponentFormat::Index16,
+                 "Invalid texture coordinates!\n(tc[i] = %d)",
+                 (u32)m_VtxDesc.high.TexCoord[i].Value());
```

Comment:
Typo here; should use `VertexComponentFormat::Direct` instead (though in practice it won't matter due to the if statement).  Not sure if the if these assertions really matter either.